src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,336 @@
+from opensearchpy import OpenSearch
+
+# Standard libraries
+from datetime import datetime, timedelta
+from shutil import copyfileobj
+from json import loads, dumps
+from os import remove, path
+import gzip
+import sys
+import csv
+
+# Adds parent directory "/sample_data_tooling" to sys.path
+sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
+from sample_data_generator.sample_data_generator import generate_data
+from sample_data_plugins.ad_plugin_data_config.data_trend_functions import AverageTrend
+
+
+def ingest_from_user_data(filename):
+    """
+    Utility function that loads user provided data to a list
+
+    Arguments:
+        - filename: The name of the user provided data file as a .json or .csv file (zipped or unzipped)
+
+    Returns:
+        - A list of data loaded from the file
+
+    Raises:
+        - TypeError: Filename is not a string or is not a valid filename
+        - ValueError: .json and .csv files are only supported
+    """
+
+    # Input validation
+    if type(filename) is not str or (type(filename) is str and "." not in filename):
+        raise TypeError("Filename is not a string or is not a valid filename")
+    name = filename.split(".gz")[0]
+    dataset = []
+
+    # Unzips file 
+    if ".gz" in name:
+        with gzip.open(name, 'rt') as fin:
+                with open(name.split(".gz")[0], 'wb') as fout:
+                    copyfileobj(fin, fout)
+    # Reads from CSV
+    if ".csv" in name:
+        with open(name, "r") as f:
+            reader = csv.reader(f)
+            fields = next(reader)
+            for row in reader:
+                entry = {}
+                for i in range(len(fields)):
+                    entry[fields[i]] = row[i]
+                dataset.append(entry)
+    # Reads from JSON 
+    elif ".json" in name:
+        with open(name, "r") as f:
+            for line in f:
+                dataset.append(loads(line))
+    else:
+        raise ValueError(".json and .csv files are only supported")
+    
+    # Removes unzipped file
+    if filename != name:
+        remove(name)
+
+    return dataset
+
+
+def ingest_validation(client = None, 
+    data_template = None,
+    index_name = None,
+    file_provided = None,  
```

Comment:
Please remove the additional whitespace here. Thanks

Reply:
Done, all trailing whitespace removed